app/controllers/api/order_cycles_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Api
+  class OrderCyclesController < BaseController
+    include EnterprisesHelper
+    respond_to :json
+
+    skip_authorization_check
+
+    def products
+      products = ProductsRenderer.new(
+        distributor,
+        order_cycle,
+        customer,
+        params.slice(:q, :page, :per_page)
+      ).products_json
+
+      render json: products
+    rescue ProductsRenderer::NoProducts
```

Comment:
This is something I never liked about the way we render the products. It feels to me that both the `ProductsRenderer` and the controller contain unnecessary code to handle the case of an empty list of products and even handled with exceptions. We should return whatever is there: either empty or not empty but always a JSON array.

Would you agree on creating an issue?